native/src/main/java/org/ballerinax/asb/util/ASBUtils.java

```diff
@@ -144,6 +149,104 @@ public static String convertString(Object value) {
         return (value == null || Objects.equals(value.toString(), "")) ? null : value.toString();
     }
 
+    /**
+     * Build the ServiceBusClientBuilder object.
+     *
+     * @param retryOptions             Retry options.
+     * @param connectionString         Connection string.
+     * @param queueName                Queue name.
+     * @param receiveMode              Receive mode.
+     * @param maxAutoLockRenewDuration Max auto lock renew duration.
+     * @param topicName                Topic name.
+     * @param subscriptionName         Subscription name.
+     * @return ServiceBusReceiverClientBuilder object.
+     */
+    public static ServiceBusReceiverClientBuilder receiverBuilder(AmqpRetryOptions retryOptions,
```

Comment:
IMO we can have a single util method (called something like `constructReceiverClient()`), which returns the `ServiceBusClient` itself (not the builder).

Reply:
and we can also add boolean parameter to the util to specify whether its should be a DLQ receiver (so that we can get rid of the `deadLetterReceiverBuilder` method)